pkg/netceptor/firewall_rules_test.go

```diff
@@ -83,3 +84,495 @@ func TestFirewallRules(t *testing.T) {
 		t.Fatal("rule #4 did not return Reject")
 	}
 }
+
+func TestParseFirewallRules(t *testing.T) {
+	tests := []struct {
+		name            string
+		rules           []FirewallRuleData
+		wantErr         bool
+		wantErrContains string
+		wantCount       int
+		testCases       []struct {
+			desc      string
+			ruleIndex int
+			data      *MessageData
+			want      FirewallResult
+		}
```

Comment:
You might want to create this struct as a stand-alone type so it can be referenced in the few places it's used. This will make it easier to change in the future, if needed.